app/code/Magento/CompareListGraphQl/Model/Resolver/DeleteCompareList.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Resolver;
+
+use Magento\Catalog\Model\CompareListFactory;
+use Magento\Catalog\Model\MaskedListIdToCompareListId;
+use Magento\Catalog\Model\ResourceModel\Product\Compare\CompareList as CompareListResource;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * Class for deleting compare list
+ */
+class DeleteCompareList implements ResolverInterface
+{
+    /**
+     * @var CompareListFactory
+     */
+    private $compareListFactory;
+
+    /**
+     * @var CompareListResource
+     */
+    private $compareListResource;
+
+    /**
+     * @var MaskedListIdToCompareListId
+     */
+    private $maskedListIdToCompareListId;
+
+    /**
+     * @param CompareListFactory $compareListFactory
+     * @param CompareListResource $compareListResource
+     * @param MaskedListIdToCompareListId $maskedListIdToCompareListId
+     */
+    public function __construct(
+        CompareListFactory $compareListFactory,
+        CompareListResource $compareListResource,
+        MaskedListIdToCompareListId $maskedListIdToCompareListId
+    ) {
+        $this->compareListFactory = $compareListFactory;
+        $this->compareListResource = $compareListResource;
+        $this->maskedListIdToCompareListId = $maskedListIdToCompareListId;
+    }
+
+    /**
+     * Remove compare list
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return Value|mixed|void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     * @throws GraphQlInputException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($args['uid'])) {
+            throw new GraphQlInputException(__('"uid" value must be specified'));
+        }
+
+        $listId = $this->maskedListIdToCompareListId->execute($args['uid']);
```

Comment:
I would add an additional check for verifying that the provided list belongs to the current customer (in case if the compare list has customer ID). It will help to avoid cases when guest can remove a compare list of registered user.

Reply:
The same suggestion is applicable for other mutations.

Reply:
**Done** Additional check was added